api/pkg/transformer/types/table/table_converter.go

```diff
@@ -24,8 +31,182 @@ func TableToJson(tbl *Table, format spec.FromTable_JsonFormat) (interface{}, err
 	}
 }
 
+func getRowIDValues(tbl *Table) []string {
+	rowIDSeries, _ := tbl.GetColumn(RowIDColumn)
+	var rowIDValues []string
+	if rowIDSeries != nil {
+		rowIDValues = rowIDSeries.Series().Records()
+	}
+	return rowIDValues
+}
+
+func convertToUPIColumns(cols []*series.Series) ([]*upiv1.Column, error) {
+	upiCols := make([]*upiv1.Column, len(cols))
+	for idx, col := range cols {
+		colName := col.Series().Name
+		upiColType := upiv1.Type_TYPE_STRING
+		switch col.Type() {
+		case series.Int:
+			upiColType = upiv1.Type_TYPE_INTEGER
+		case series.Float:
+			upiColType = upiv1.Type_TYPE_DOUBLE
+		case series.String:
+			upiColType = upiv1.Type_TYPE_STRING
+		default:
+			return nil, fmt.Errorf("type %v is not supported in UPI", col.Type())
+		}
+
+		upiCols[idx] = &upiv1.Column{
+			Name: colName,
+			Type: upiColType,
+		}
+	}
+	return upiCols, nil
+}
+
+// ToUPITable converts table into upi table
+// will exclude row_id columns from upi columns
+func ToUPITable(tbl *Table, name string) (*upiv1.Table, error) {
+	cols := tbl.ColumnsWithExcluding([]string{RowIDColumn})
+	upiCols, err := convertToUPIColumns(cols)
+	if err != nil {
+		return nil, err
+	}
+
+	rowIDValues := getRowIDValues(tbl)
+	colValues, err := getTableRecordsForColumns(cols)
+	if err != nil {
+		return nil, err
+	}
+
+	upiRows := make([]*upiv1.Row, tbl.NRow())
+	rowIDColExist := tbl.NRow() == len(rowIDValues)
+	for rowIdx := 0; rowIdx < tbl.NRow(); rowIdx++ {
+		vals := make([]*upiv1.Value, len(cols))
+		for colIdx, col := range upiCols {
+			colValues := colValues[col.Name]
```

Comment:
This variable name is shadowing `colValues` declared outside the block. Can you rename it to avoid confusion or bug?